src/base/memory_monitor_sr.cpp

```diff
@@ -281,61 +315,108 @@ namespace cubperf
 
 	    if (!comp_skip)
 	      {
-		comp_idx = add_component (info[cnt].comp_name);
+		comp_idx = add_component (info[idx].comp_name);
 	      }
-	    /* XXX: add error if comp_idx > MAX_COMP_IDX */
+	    assert (comp_idx < MAX_COMP_IDX);
 
-	    if (info[cnt].subcomp_name)
+	    if (info[idx].subcomp_name)
 	      {
-		subcomp_idx = m_component[comp_idx]->get_subcomp_index (info[cnt].subcomp_name);
+		subcomp_idx = m_component[comp_idx]->get_subcomp_index (info[idx].subcomp_name);
 		if (subcomp_idx != -1)
 		  {
 		    subcomp_skip = true;
 		  }
 
 		if (!subcomp_skip)
 		  {
-		    subcomp_idx = m_component[comp_idx]->add_subcomponent (info[cnt].subcomp_name);
+		    subcomp_idx = m_component[comp_idx]->add_subcomponent (info[idx].subcomp_name);
 		  }
-		/* XXX: add error if subcomp_idx > MAX_SUBCOMP_IDX */
+		assert (subcomp_idx < MAX_COMP_IDX);
 	      }
+	    else
+	      {
+		assert (info[idx].subcomp_name == NULL);
+	      }
+	  }
+	else
+	  {
+	    assert (info[idx].comp_name == NULL);
 	  }
 
-	m_comp_idx_map.push_back (make_stat_idx_map (comp_idx, subcomp_idx));
+	m_comp_idx_map.push_back (make_comp_idx_map_val (comp_idx, subcomp_idx));
 
-	cnt++;
+	idx++;
       }
   }
 
   int mmon_module::aggregate_stats (const MMON_MODULE_INFO &info)
   {
-    return 0;
+    int error = NO_ERROR;
+    return error;
   }
 
-  void mmon_module::add_stat (uint64_t size, int comp_idx, int subcomp_idx, bool init, bool expand)
+  void mmon_module::add_stat (uint64_t size, int comp_idx, int subcomp_idx, bool allocation_at_init, bool is_expanded)
```

Comment:
Is it just for `move_stat`? Or is there a meaningful use case for a user?

Reply:
is_expanded is used to update expand_count. It can be occurred in resize_stat.

Reply:
I can see that it happens in `resize_stat`. What I'm wondering is whether it can't be counted automatically. It seems that every add_stat should be called with `is_expanded = true` except for the move_stat. So, I'd like to know another case that a user can do. I want to relate to the need of this.